class.jetpack-sync.php

```diff
@@ -688,6 +688,10 @@ function options( $file, $option /*, $option, ... */ ) {
 			$this->sync_options[$module_slug] = array();
 		}
 
+		if ( isset( $options[0] ) && is_array( $options[0] ) ) {
```

Comment:
I would remove this. It is not needed, ( the for loop you made created below solved this issue) and can cause confusion. For example why just the second parameter? 

If you would want it to work this way. I think you should use some sort of array merge.